package/MDAnalysis/analysis/pca.py

```diff
@@ -230,6 +231,11 @@ class PCA(AnalysisBase):
        ``mean`` input now accepts coordinate arrays instead of atomgroup.
        :attr:`p_components`, :attr:`variance` and :attr:`cumulated_variance`
        are now stored in a :class:`MDAnalysis.analysis.base.Results` instance.
+    .. versionchanged:: 2.8.0
+       ``self.run()`` can now appropriately use ``frames`` parameter (bug
+       described by #4425 and fixed by #4423). Previously, behaviour was to
+       manually iterate through ``self._trajectory``, which starting #3415 that
+       introduced ``frames`` is wrong, and should use ``self._sliced_trajectory``.
```

Comment:
```suggestion
       manually iterate through ``self._trajectory``, which would
       incorrectly handle cases where the ``frame`` argument
       was passed.
```
Just a slight personal preference edit - I always see the versionchanged as intended for users, so what they want to know is "what changed that could have affected my results" rather than "what internal code changes were necessary".